lib/rubocop/cop/rspec/capybara/specific_finders.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Capybara
+        # Checks for there is a more specific finder offered by Capybara.
+        #
+        # @example
+        #   # bad
+        #   find('#some-id')
+        #   find('[visible][id=some-id]')
+        #
+        #   # good
+        #   find_by_id('some-id')
+        #   find_by_id('some-id', visible: true)
+        #
+        class SpecificFinders < Base
+          extend AutoCorrector
+
+          include CssSelector
+          include RangeHelp
+
+          MSG = 'Prefer `find_by` over `find`.'
+          RESTRICT_ON_SEND = %i[find].freeze
+
+          # @!method find_argument(node)
+          def_node_matcher :find_argument, <<~PATTERN
+            (send _ :find (str $_)...)
+          PATTERN
+
+          def on_send(node)
+            find_argument(node) do |arg|
+              next if multiple_selectors?(arg)
+
+              on_attr(node, arg) if attribute?(arg) && common_attributes?(arg)
+              on_id(node, arg) if id?(arg)
+            end
+          end
+
+          private
+
+          def on_attr(node, arg)
+            id = attribute_value(arg, 'id')
+            return if id.empty?
+
+            add_offense(offense_range(node)) do |corrector|
+              replace_good_method(corrector, node)
+
+              corrector.replace(node.first_argument.loc.expression,
+                                replaced_argments(arg, id))
+            end
+          end
+
+          def on_id(node, arg)
+            add_offense(offense_range(node)) do |corrector|
+              replace_good_method(corrector, node)
+              corrector.replace(node.first_argument.loc.expression,
+                                "'#{arg.delete('#')}'")
+            end
+          end
```

Comment:
Perhaps those could be DRYed

```suggestion
          def on_attr(node, arg)
            id = attribute_value(arg, 'id')
            return if id.empty?

            register_offense(node, replaced_argments(arg, id))
          end

          def on_id(node, arg)
            register_offense(node, arg.to_s.delete('#'))
          end
          
          def register_offense(node, arg_replacemenet)
            add_offense(offense_range(node)) do |corrector|
              replace_good_method(corrector, node)
              corrector.replace(node.first_argument.loc.expression, arg_replacemenet)
            end
          end
```